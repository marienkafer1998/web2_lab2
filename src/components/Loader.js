import React from "react";


export default function Loader(props) {
    return (
        <div>
            {/*<img src={"https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"} alt="loader" />*/}
            <p>Weather is loading...</p>
        </div>
    );
}