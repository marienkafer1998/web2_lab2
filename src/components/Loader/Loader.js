import React from "react";
import {Container} from "reactstrap";


export default function Loader(props) {
    return (
        <Container>

            <h1 >Please, wait, weather is loading...</h1>


            <img src={"https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"} alt="Weather is loading" width={"100px"} />

        </Container>
    );
}