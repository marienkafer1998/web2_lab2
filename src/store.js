import { createStore, combineReducers, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { LOCAL_STORAGE_KEY } from "./LocalStorage";

import geoReducer from "./redux/geoReducer";
import favReducer from "./redux/favReducer";

const store = createStore(
    combineReducers({geo: geoReducer, fav: favReducer}),
    applyMiddleware(logger, thunk)
);



store.subscribe(() => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify([...store.getState().fav.favorites.keys()]));
});

export default store;
