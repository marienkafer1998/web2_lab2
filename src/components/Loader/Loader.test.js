import React from 'react';
import renderer from 'react-test-renderer';
import Loader from "./Loader.js";
import {Provider} from "react-redux";
import store from "../../store";

it('Loader functional rendered properly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Loader/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

