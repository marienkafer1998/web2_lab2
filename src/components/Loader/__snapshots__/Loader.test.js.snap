// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader functional rendered properly 1`] = `
<div
  className="container"
>
  <h1>
    Please, wait, weather is loading...
  </h1>
  <img
    alt="Weather is loading"
    src="https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"
    width="100px"
  />
</div>
`;
